pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,41 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, certificate *corev1.LocalObjectReference, clusterDomain string) error {
+	clusterHasManagedDomain := clusterHasManagedDomain(clusterDomain)
+
+	if certificate == nil {
+		if clusterHasManagedDomain {
+			return fmt.Errorf("ARO has not provisioned the default ingress certificate because it is empty and cluster has a managed domain")
```

Comment:
does it ever happen? wouldn't we have errors beforehand? like, errors when we create the cluster?

Reply:
I think it is necessary, because either:
- if ARO could fail at provisionning the certifcate during the 			steps.Action(m.configureIngressCertificate) step in installation.  Cluster installation will fail (provisionning failed), but the cluster would still be there and mostly operational (this step is one of the last steps). This is an ARO problem in that case.
- customer could have deleted the cert. You are right, that one is also a possibility. Maybe we should mention the two possibilities

Reply:
As far as I understood when talking with @petrkotas, we have to differentiate those errors into different types because one case is the customer's fault and the other is ARO's fault. Even if the procedure is the same, we will have a better understanding of the situation). 

Agree on the rewording of the error message.
Thank you both!